packages/babel-traverse/src/path/inference/inferer-reference.ts

```diff
@@ -186,7 +186,7 @@ function inferAnnotationFromBinaryExpression(
 function getParentConditionalPath(
   binding: Binding,
   path: NodePath,
-  name: string,
+  name: string | undefined,
```

Comment:
The `name` here is passed from a `name` property of an `Identifier`, I think it should be always a string unless the AST is malformed.

https://github.com/babel/babel/blob/c36fa6a9299df95b801695736f78ac87211f009f/packages/babel-traverse/src/path/inference/inferer-reference.ts#L23

https://github.com/babel/babel/blob/c36fa6a9299df95b801695736f78ac87211f009f/packages/babel-traverse/src/path/inference/inferer-reference.ts#L55

Can we remove the `optional` marker here?

https://github.com/babel/babel/blob/c36fa6a9299df95b801695736f78ac87211f009f/packages/babel-traverse/src/path/inference/inferer-reference.ts#L211